src/CoreWCF.NetTcp/src/CoreWCF/Channels/NetTcpOptions.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System;
+
+namespace CoreWCF.Channels
+{
+    public class NetTcpOptions
+    {
+        internal List<TcpListenOptions> CodeBackedListenOptions { get; } = new List<TcpListenOptions>();
+        public IServiceProvider ApplicationServices { get; internal set; }
```

Comment:
Does this getter require to be public ?

Reply:
Yes, it mirrors how Kestrel handles this and enables you to pull things from DI in your delegate. E.g. you can do this:
```c#
webHost.UseNetTcp((NetTcpOptions options) =>
{
    IConfiguration configuration = options.ApplicationServices.GetRequiredService<IConfiguration>();
    // pull base address from config
    options.Listen(baseAddressFromConfig);
});
```